phoenix-core/src/main/java/org/apache/phoenix/log/QueryLogger.java

```diff
@@ -92,9 +92,17 @@ public boolean isSynced(){
         }
     };
 
-    public static QueryLogger getInstance(PhoenixConnection connection, boolean isSystemTable) {
-        if (connection.getLogLevel() == LogLevel.OFF || isSystemTable || ThreadLocalRandom.current()
-                .nextDouble() > connection.getLogSamplingRate()) { return NO_OP_INSTANCE; }
+    public static QueryLogger getInstance(PhoenixConnection connection, boolean isSystemTable,
+        boolean criticalStatement) {
+        // always log critical statements (DROP,ALTER for now)
+        // do not log anything when loglevel is off.
+        // do not log systemTable statement.
+        // do sampling on other statements based on configured perncentage, 1% by default.
+        if ((connection.getLogLevel() == LogLevel.OFF || isSystemTable
+            || ThreadLocalRandom.current().nextDouble() > connection.getLogSamplingRate()) &&
+            !criticalStatement) {
```

Comment:
Does this mean that it will log the statements even if the feature is disabled on server side? Can you add a test for that? Do we really want it that way?

Reply:
Changed, do not log when loglevel is OFF>